apps/beeai-server/src/beeai_server/utils/github.py

```diff
@@ -82,3 +110,57 @@ def __str__(self):
         version = f"@{self.version}" if self.version else ""
         path = f"#path={self.path}" if self.path else ""
         return f"git+https://github.com/{self.org}/{self.repo}{version}{path}"
+
+
+def _is_unsafe_file(member: tarfile.TarInfo) -> bool:
+    filename = member.name
+    if (
+        filename.startswith("/")
+        or ".." in filename
+        or not pathlib.Path(filename).absolute().is_relative_to(pathlib.Path(".").resolve())
+    ):
+        return True
+    return False
+
+
+def extract_targz_safe(tar_path: PathLike | Path, extract_path: PathLike | Path):
+    with tarfile.open(tar_path, "r:gz") as tar:
+        members = []
+        for member in tar.getmembers():
+            if member.linkpath or member.linkname:
+                logger.warning(f"Cannot extract member: {member.name}, symlinks are forbidden")
```

Comment:
Actually what's the threat model here? We run the executables with no sandboxing anyway 😁

Reply:
I think you can unzip to some unwanted paths, like replace executable in `~/local/.bin` :D tar allows both absolute paths (from fs root) and relative paths (`../../`) and links, see this big red warning :D 

https://docs.python.org/3/library/tarfile.html#tarfile.TarFile.extractall

Zip does not have this issue, but it does not preserve fs permissions and we need some files executable